Core/src/IfcGeometryConverter/SolidModelConverter.h

```diff
@@ -2259,126 +2258,119 @@ namespace OpenInfraPlatform
 
-					int var = 0;
+						std::vector<std::shared_ptr<carve::input::PolylineSetData>> surfacePolylineData = surfaceData->polylines;
+						std::vector<std::vector<carve::geom::vector<3>>> baseSurfacePointVector;
 
-					if (var == 0)
-					{
-						std::shared_ptr<carve::input::PolylineSetData> surface_data =
-							faceConverter->convertIfcSurface(base_surface, carve::math::Matrix::IDENT());
-						std::vector<carve::geom::vector<3>> base_surface_points = surface_data->points;
+						for (auto polyline : surfacePolylineData) {
+							baseSurfacePointVector.push_back(polyline->points);
+						}
+
+						std::vector<carve::geom::vector<3>> baseSurfacePoints = baseSurfacePointVector[0];
 
-						if (base_surface_points.size() != 4)
+						if (baseSurfacePoints.size() != 4)
 						{
-							BLUE_LOG(warning) << "Invalid IfcHalfSpaceSolid.BaseSurface #" << base_surface->getId();
-							return;
+							throw oip::InconsistentModellingException(halfSpaceSolid, "Invalid BaseSurface selected!");
 						}
+
 						// If the agreement flag is TRUE, then the subset is the one the normal points away from
-						bool agreement = halfSpaceSolid->AgreementFlag;
-						if (!agreement)
+						if (!halfSpaceSolid->AgreementFlag)
 						{
-							std::reverse(base_surface_points.begin(), base_surface_points.end());
+							std::reverse(baseSurfacePoints.begin(), baseSurfacePoints.end());
 						}
-						carve::geom::vector<3>  base_surface_normal = GeomUtils::computePolygonNormal(base_surface_points);
+						carve::geom::vector<3>  baseSurfaceNormal = GeomUtils::computePolygonNormal(baseSurfacePoints);
 
-						carve::geom::vector<3>  half_space_extrusion_direction = -base_surface_normal;
-						carve::geom::vector<3>  half_space_extrusionVector = half_space_extrusion_direction * HALF_SPACE_BOX_SIZE;
-						std::shared_ptr<carve::input::PolyhedronData> half_space_box_data(new carve::input::PolyhedronData());
-						itemData->closed_polyhedrons.push_back(half_space_box_data);
-						extrudeBox(base_surface_points, half_space_extrusionVector, half_space_box_data);
+						carve::geom::vector<3>  halfSpaceExtrusionDirection = -baseSurfaceNormal;
+						carve::geom::vector<3>  halfSpaceExtrusionVector = halfSpaceExtrusionDirection * HALF_SPACE_BOX_SIZE;
+						std::shared_ptr<carve::input::PolyhedronData> halfSpaceBoxData(new carve::input::PolyhedronData());
+						itemData->closed_polyhedrons.push_back(halfSpaceBoxData);
+						extrudeBox(baseSurfacePoints, halfSpaceExtrusionVector, halfSpaceBoxData);
 
 						// apply object coordinate system
-						for (std::vector<carve::geom::vector<3> >::iterator it_points = half_space_box_data->points.begin(); it_points != half_space_box_data->points.end(); ++it_points)
+						for ( auto point : halfSpaceBoxData->points)
 						{
-							carve::geom::vector<3> & poly_point = (*it_points);
-							poly_point = pos * poly_point;
+							carve::geom::vector<3> & polyPoint = pos * point;
 						}

```

Comment:
This changed the previous meaning.

It should say

for( auto& point : halfSpaceBoxData->points )
point = pos * point;

Reply:
Resolved: 01cd5dc